stock_measuring_device/wizard/measuring_wizard.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+from odoo import api, fields, models
+
+
+class MeasuringWizard(models.TransientModel):
+    _name = "measuring.wizard"
+    _inherit = "barcodes.barcode_events_mixin"
+    _description = "measuring Wizard"
+    _rec_name = "device_id"
+
+    product_id = fields.Many2one("product.product", domain=[("type", "=", "product")])
+    line_ids = fields.One2many("measuring.wizard.line", "wizard_id")
+    device_id = fields.Many2one("measuring.device", readonly=True)
+
+    @api.onchange("product_id")
+    def onchange_product_id(self):
+        if self.product_id:
+            to_create = []
+            to_create += self._prepare_unit_line()
+            to_create += self._prepare_packaging_lines()
+            recs = self.env["measuring.wizard.line"].create(to_create)
+            self.line_ids = recs
+        else:
+            self.line_ids = [(5, 0, 0)]
+
+    def _prepare_unit_line(self):
+        vals = {
+            "wizard_id": self.id,
+            "sequence": 0,
+            "name": "Unit",
+            "qty": 1,
+            "max_weight": self.product_id.weight,
+            "lngth": self.product_id.product_length,
+            "width": self.product_id.product_width,
+            "height": self.product_id.product_height,
+        }
+        product_dimension_uom = self.product_id.dimensional_uom_id
+        mm_uom = self.env.ref("stock_measuring_device.product_uom_mm")
+        if mm_uom != product_dimension_uom:
+            vals.update(
+                {
+                    "lngth": product_dimension_uom._compute_quantity(
+                        self.product_id.product_length, mm_uom
+                    ),
+                    "width": product_dimension_uom._compute_quantity(
+                        self.product_id.product_width, mm_uom
+                    ),
+                    "height": product_dimension_uom._compute_quantity(
+                        self.product_id.product_height, mm_uom
+                    ),
+                }
+            )
+        return [vals]
+
+    def _prepare_packaging_lines(self):
+        vals_list = []
+        product_packaging = self.env["product.packaging"]
+        packaging_types = self.env["product.packaging.type"].search([])
```

Comment:
maybe a silly question as I miss a bit of context, but... this search and the following loop could generate tons of wiz lines, no? Is there any way to limit this?

Reply:
you typically don't have that many packaging types: unit, box, retail box, pallet. Give or take a few. So no.